test/e2e/subscription_e2e_test.go

```diff
@@ -1253,6 +1253,141 @@ var _ = Describe("Subscription", func() {
+				},
+				DefaultChannelName: stableChannel,
+			},
+			{
+				PackageName: packageName2,
+				Channels: []registry.PackageChannel{
+					{Name: stableChannel, CurrentCSVName: csvB.GetName()},
+				},
+				DefaultChannelName: stableChannel,
+			},
+		}
+
+		manifests2 := []registry.PackageManifest{
+			{
+				PackageName: packageName2,
+				Channels: []registry.PackageChannel{
+					{Name: stableChannel, CurrentCSVName: csvC.GetName()},
+				},
+				DefaultChannelName: stableChannel,
+			},
+		}
+
+		catalogSourceName := genName("catsrc")
+		catsrc, cleanup := createInternalCatalogSource(kubeClient, crClient, catalogSourceName, testNamespace, manifests, []apiextensions.CustomResourceDefinition{crd, crd2}, []v1alpha1.ClusterServiceVersion{csvA, csvABC, csvB})
+		defer cleanup()
+
+		// Ensure that the catalog source is resolved before we create a subscription.
+		_, err := fetchCatalogSourceOnStatus(crClient, catsrc.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+		require.NoError(GinkgoT(), err)
+
+		subscriptionSpec := &v1alpha1.SubscriptionSpec{
+			CatalogSource:          catsrc.GetName(),
+			CatalogSourceNamespace: catsrc.GetNamespace(),
+			Package:                packageName1,
+			Channel:                stableChannel,
+			StartingCSV:            csvA.GetName(),
+			InstallPlanApproval:    v1alpha1.ApprovalAutomatic,
+		}
+
+		catalogSourceName2 := genName("catsrc")
+		catsrc2, cleanup2 := createInternalCatalogSource(kubeClient, crClient, catalogSourceName2, testNamespace, manifests2, []apiextensions.CustomResourceDefinition{crd2}, []v1alpha1.ClusterServiceVersion{csvC})
+		defer cleanup2()
+
+		// Ensure that the catalog source is resolved before we create a subscription.
+		_, err = fetchCatalogSourceOnStatus(crClient, catsrc2.GetName(), testNamespace, catalogSourceRegistryPodSynced)
+		require.NoError(GinkgoT(), err)
+
+		// Create a subscription that has a dependency
+		subscriptionName := genName("sub-")
+		cleanupSubscription := createSubscriptionForCatalogWithSpec(GinkgoT(), crClient, testNamespace, subscriptionName, subscriptionSpec)
+		defer cleanupSubscription()
+
+		subscription, err := fetchSubscription(crClient, testNamespace, subscriptionName, subscriptionStateAtLatestChecker)
+		require.NoError(GinkgoT(), err)
+		require.NotNil(GinkgoT(), subscription)
+
+		// Check that a single catalog source was used to resolve the InstallPlan
+		_, err = fetchInstallPlan(GinkgoT(), crClient, subscription.Status.InstallPlanRef.Name, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete))
+		require.NoError(GinkgoT(), err)
+		// Fetch CSVs
+		_, err = fetchCSV(GinkgoT(), crClient, csvB.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
+		_, err = fetchCSV(GinkgoT(), crClient, csvC.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
+		_, err = fetchCSV(GinkgoT(), crClient, csvA.Name, testNamespace, csvSucceededChecker)
+		require.NoError(GinkgoT(), err)
```

Comment:
I realize this should fail resolution, but can you add one negative check to make sure ABC doesn't exist?